party_time/party_time.lua

```diff
@@ -0,0 +1,404 @@
+    options.whitelist[sub_cmd](options.whitelist, names)
+    settings.save(options)
+end
+
+pt:register('w', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+pt:register('whitelist', whitelist, '<sub_cmd:lookup_add_remove> {name}*')
+
+
+local ui_enable = function(bool)
+    options.ui.enabled = bool
+    settings.save(options)
+end
+
+pt:register('ui_enable', ui_enable, '<enabled:lookup_boolean>')
+
+
+local auto_accept_enable = function(bool)
+    options.auto.accept = bool
+    settings.save(options)
+end
+
+pt:register('auto_accept', auto_accept_enable, '<enabled:lookup_boolean>')
+
+
+local auto_decline_enable = function(bool)
+    options.auto.decline = bool
+    settings.save(options)
+end
+
+pt:register('auto_decline', auto_decline_enable, '<enabled:lookup_boolean>')
+
+
+local default_handler = function(option)
+    options.default = option
+    settings.save(options)
+end
+
+pt:register('defualt', default_handler, '<option:lookup_option>')
+
+-- Packet Event Handlers
+-- Recieve Invites & Recieve Requests
+local default_handlers = {
+    invite = {
+        ask = function(name)
+            invite_dialog = {
+                state = invite_dialog_state,
+                add_to_whitelist = false,
+                name = name,
+            }
+            unhandled_invite = true
+        end,
+        whitelist = function()
+            coroutine.schedule(function()
+                local clock = os.clock()
+                repeat
+                    if (os.clock() - clock) > 90 then
+                        return
+                    end
+                    coroutine.sleep_frame()
+                until(#treasure == 0)
+                command.input('/join')
+            end)
+        end,
+        blacklist = function()
+            command.input('/decline')
+        end
+    },
+    request = { 
+        ask = function(name)
+            unhandled_requests[name] = {
+                state = {
+                    title = 'Party Request',
+                    style = 'normal',
+                    x = options.ui.x,
+                    y = options.ui.y,
+                    width = 179,
+                    height = 96,
+                    resizable = false,
+                    moveable = true,
+                    closable = true,
+                },
+                add_to_whitelist = false,
+            }
+        end,
+        whitelist = function(name)
+            command.input('/pcmd add '.. name)
+        end,
+        blacklist = function()
+            return
+        end,
+    }
+}
+
+packets.incoming[0x0DC]:register(function(p)
+    if options.auto.accept and options.whitelist:contains(p.player_name) then
+        coroutine.schedule(function()
+            local clock = os.clock()
+            repeat
+                if (os.clock() - clock) > 90 then
+                    return
+                end
+                coroutine.sleep_frame()
+            until(#treasure == 0)
+            command.input('/join')
```

Comment:
We could add a UI popup to notify the user that the addon is waiting for the pool to clear. This should probably not be in this release, but can add it to the to-do list. It always bothered me in v4 that *AutoJoin* sometimes didn't auto-join and I always forgot why, until I checked the treasure icon...